controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,169 @@
+    int timeStep = int(supervisor.getBasicTimeStep());
+    
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+    
+    // Generate distributions for random number systems in the following loop
+    std::uniform_int_distribution<> xDistrib(0, 860);
+    std::uniform_int_distribution<> yDistrib(0, 580);
+    std::uniform_int_distribution<> rotDistrib(0, rotations.size() - 1);
+
+    while (supervisor.step(timeStep) != -1) {
+        
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
+        // Loop through every robot
+        for (long unsigned int i = 0; i < otherRobotsNodes.size(); i++)
+        {
+          // Assume there is no collision
+          bool collision = false;
+          // newPos will be a "Proposed location" for a robot to teleport to
+          std::array<double, 3> newPos;
+          do
+          {
+            collision = false;
+            // Generate a new random location
+            newPos[0] = 4.3 - xDistrib(gen) / 100;
+            newPos[1] = 2.9 - yDistrib(gen) / 100;
+            newPos[2] = 0.51;
+            
+            // Loop through the vector of existing proposed locations and see if the new one is going to
+            // collide with any of them
+            for (std::array<double, 3> testPos : positions)
+            {
+              double distance = sqrt(pow((newPos[1] - testPos[1]), 2) + pow((newPos[0] - testPos[0]), 2));
+              if (distance < 0.1)
+              {
+                collision = true;
+              } 
+            }
+          }
+          // Loop until a proposed location has been found that doesn't clash with the existing ones
+          while (collision); 
+          // Finally add the proposed location in as a confirmed position
+          positions.push_back(newPos);
+        }
+
+
+        // Loop through every robot
+        for (long unsigned int i = 0; i < otherRobotsNodes.size(); i++)
+        {
+           // Grab translation field of the robot to modify
+           webots::Field& target_translation_field = *(otherRobotsNodes[i])->getField("translation");
+           // Convert the field to a vector to output to console
+           const double* target_translation_vec = target_translation_field.getSFVec3f();
+           
+           // Grab the current rotation field of the robot to modify
+           webots::Field& target_rotation_field = *(otherRobotsNodes[i])->getField("rotation");
+          // Convert the field to a vector to output to console
+          const double* target_rotation_vec = target_rotation_field.getSFRotation();
+          
+          // Output current location/rotation
+          std::ofstream log;
```

Comment:
Use yaml to log?

Reply:
Do you want it as a list of like [time, location, rotation]?

Reply:
Something like 
```yaml
# Simulation time
time: t
# Current robot position
location: [x,y,z]
# Angle axis rotation x,y,z,alpha
rotation: [x,y,z,alpha]
```

Reply:
As an entry in a list though? Won't it overwrite with every new iteration of the loop if I just have them in the YAML as that?

Reply:
Hmmm I guess then the question is why are we logging it? It might make more sense if each robot logs it's own information itself. If we want this information with the vision data information for eg localisation, then it should be happening when we get the images, ie in a different controller.

Reply:
As long as those other controllers have a way to output to a different text file each - maybe have to pass in a filename as an argument... But yea just remove this bit then?

Reply:
Yeah that's what the vision data collection controller does, since the visual mesh wants individual yaml files that correspond to image/s. Probs remove it yeah unless someone else thinks otherwise.